Create wdio reporter (#32)
cerner/terra-toolkit-boneyard: docs/TerraWDIOSpecReporter.md

# Wdio File Reporter

> Wdio File Reporter is a reporter that logs wdio test output to separate files based on locale, theme and form-factor with the following attributes

Reviewer: ```suggestion
Wdio File Reporter is a reporter that logs wdio test output to separate files based on locale, theme, and form-factor with the following attributes
```

- date and time of when the tests are started and when they are finished

> - Locale, Form Factor and Theme of the tests 

Reviewer: ```suggestion
- Locale, Form Factor, and Theme of the tests 
```

- Name of the tests and whether they've succeeded or failed

**WDIO test output Directory**

> Terra WDIO spec Reporter logs the WDIO test output in tests/wdio/reports/ or test/wdio/reports/ depending on whether tests or test is the directory with tests

Reviewer: ```suggestion
Terra WDIO spec Reporter logs the WDIO test output in tests/wdio/reports/ or test/wdio/reports/ depending on whether tests or test is the directory that contains specs
```

**Check for Mono Repo**
Terra WDIO spec Reporter assumes mono-repo will have packages directory in the root folder

## Usage

Add WdioFileReporter as an additional reporter within the wdio.config file. Include "dot" to avoid overriding default reporters

```javascript

const WdioFileReporter = require('terra-toolkit/src/wdio/WdioFileReporter');
{
 reporters: ['dot', WdioFileReporter],
}

```
## Report Format
- The name of the log file will be **result-\<locale>-\<theme>-\<form-factor>-\<browser>-\<repo-name>.json**(eg: result-en-huge-chrome-terra-toolkit-boneyard.json)

> ```

Reviewer: this might be out of scope, but on second thought, we could convert this to an `mdx` file, generate these examples, then import the json instead of hardcoding the example.

# Wdio File Reporter

Wdio File Reporter is a reporter that logs wdio test output to separate files based on locale, theme and form-factor with the following attributes

- date and time of when the tests are started and when they are finished

- Locale, Form Factor and Theme of the tests 

- Name of the tests and whether they've succeeded or failed

> ***WDIO test output Directory ***

Reviewer 2: ```suggestion
**WDIO test output Directory**
```

Terra WDIO spec Reporter logs the WDIO test output in tests/wdio/reports/results or test/wdio/reports/results depending on whether tests or test is the directory with tests

> ***Check for Mono Repo**

Reviewer 2: ```suggestion
**Check for Mono Repo**
```

Terra WDIO spec Reporter assumes mono-repo will have packages directory in the root folder

## Usage

Add WdioFileReporter as an additional reporter within the wdio.config file. Include "dot" to avoid overriding default reporters

```javascript

const WdioFileReporter = require('terra-toolkit/src/wdio/WdioFileReporter');
{
 reporters: ['dot', WdioFileReporter],
}

```
## Report Format

> - The name of the log file will be **result-\<locale>-\<theme>-\<form-factor>.json**(eg: result-en-huge.json)

Reviewer 2: Is this still accurate?

```
{
  "startDate": "4/29/2020, 2:24:35 PM",
  "type": "wdio",
  "locale": "en",
  "formFactor": "tiny",
  "theme": "default-theme",
  "Output": [
    [
      "------------------------------------------------------------------",
      "[chrome #0-0] Session ID: e0217cb2b64dc2598599c1b200b6e4ff",
      "[chrome #0-0] Spec: /Users/sn081183/Desktop/terra-toolkit-boneyard/tests/wdio/i18n-spec.js",
      "[chrome #0-0] Running: chrome",
      "[chrome #0-0]",
      "[chrome #0-0] I18n Locale",
      "[chrome #0-0]   ✓ Express correctly sets the application locale",
      "[chrome #0-0]   ✓ [default] to be within the mismatch tolerance",
      "[chrome #0-0]",
      "[chrome #0-0]",
      "[chrome #0-0] 2 passing (3s)",
      "[chrome #0-0]",
      "",

>       ""

Reviewer 2: are these empty strings still printed?

Author: yes

Author: now this should not give any empty strings 
https://github.com/cerner/terra-toolkit-boneyard/blob/931b0e2cfcac32d044ae84fc18e51daa9286b815/reporters/wdio/TerraWDIOSpecReporter.js#L147

- date and time of when the tests are started and when they are finished

- Locale, Form Factor, and Theme of the tests 

- Name of the tests and whether they've succeeded or failed

**WDIO test output Directory**
Terra WDIO spec Reporter logs the WDIO test output in tests/wdio/reports/ or test/wdio/reports/ depending on whether tests or test is the directory that contains specs

**Check for Mono Repo**
Terra WDIO spec Reporter assumes mono-repo will have packages directory in the root folder

## Usage

Add WdioFileReporter as an additional reporter within the wdio.config file. Include "dot" to avoid overriding default reporters

```javascript

const WdioFileReporter = require('terra-toolkit/src/wdio/WdioFileReporter');
{
 reporters: ['dot', WdioFileReporter],
}

```
## Report Format
- The name of the log file for non-monorepo will be **result-\<locale>-\<theme>-\<form-factor>-\<browser>-\<repo-name>.json**(eg: result-en-huge-chrome-terra-toolkit-boneyard.json)

- The name of the log file for mono-repo will be **result-\<locale>-\<theme>-\<form-factor>-\<browser>-<Package-name>.json**(eg: result-clinical-lowlight-theme-chrome-terra-clinical-data-grid.json)

> - This is output example [a relative link](TerraWDIOSpecReporter-sample-result.json)

Reviewer: ```suggestion
- Example output [a relative link](TerraWDIOSpecReporter-sample-result.json)
```

i'm fine with this, but my previous comment was about converting this file to mdx, then importing the file 

```
import sample-result.json
```

# Wdio File Reporter

Wdio File Reporter is a reporter that logs wdio test output to separate files based on locale, theme and form-factor with the following attributes

- date and time of when the tests are started and when they are finished

- Locale, Form Factor and Theme of the tests 

- Name of the tests and whether they've succeeded or failed

## Usage

Add WdioFileReporter as an additional reporter within the wdio.config file. Include "dot" to avoid overriding default reporters

```javascript

const WdioFileReporter = require('terra-toolkit/src/wdio/WdioFileReporter');
{
 reporters: ['dot', WdioFileReporter],
}

> ```

Reviewer 2: We should add info on adding this file to the .gitIgnore, .dockerIgnore, and .npmIgnore files in a project
